bpf/gotracer/go_nethttp.c

```diff
@@ -270,21 +294,44 @@ int beyla_uprobe_readContinuedLineSliceReturns(struct pt_regs *ctx) {
     u64 len = (u64)GO_PARAM2(ctx);
     u8 *buf = (u8 *)GO_PARAM1(ctx);
 
-    if (len >= (W3C_KEY_LENGTH + W3C_VAL_LENGTH + 2)) {
-        u8 temp[W3C_KEY_LENGTH + W3C_VAL_LENGTH + 2];
-        bpf_probe_read(temp, sizeof(temp), buf);
-        bpf_dbg_printk("goroutine_addr %lx", goroutine_addr);
-        go_addr_key_t g_key = {};
-        go_addr_key_from_id(&g_key, goroutine_addr);
+    u8 temp[HTTP_HEADER_MAX_LEN];
+    if (len > sizeof(temp))
+        len = sizeof(temp);
+    bpf_probe_read(temp, len, buf);
+
+    bpf_dbg_printk("goroutine_addr %lx", goroutine_addr);
+    go_addr_key_t g_key = {};
+    go_addr_key_from_id(&g_key, goroutine_addr);
 
-        connection_info_t *existing = bpf_map_lookup_elem(&ongoing_server_connections, &g_key);
-        if (existing) {
-            if (!bpf_memicmp((const char *)temp, "traceparent: ", W3C_KEY_LENGTH + 2)) {
-                server_http_func_invocation_t inv = {};
-                decode_go_traceparent(
-                    temp + W3C_KEY_LENGTH + 2, inv.tp.trace_id, inv.tp.parent_id, &inv.tp.flags);
-                bpf_dbg_printk("Found traceparent in header %s", temp);
-                bpf_map_update_elem(&ongoing_http_server_requests, &g_key, &inv, BPF_ANY);
+    int w3c_value_start = W3C_KEY_LENGTH + 2; // "traceparent: "
+    int w3c_header_length = w3c_value_start + W3C_VAL_LENGTH;
+    int content_type_value_start = CONTENT_TYPE_KEY_LEN + 2; // "content-type: "
+    int content_type_header_length = content_type_value_start + HTTP_CONTENT_TYPE_MAX_LEN;
+
+    connection_info_t *existing = bpf_map_lookup_elem(&ongoing_server_connections, &g_key);
+    if (existing) {
```

Comment:
nit:

```suggestion
    if (!existing) {
        return 0;
    }
```
rationale: keeps indentation at bay -> improve readability (our code is guilty of this elsewhere)